server/apiserver/argoserver.go

```diff
@@ -303,7 +304,15 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 	mustRegisterGWHandler(workflowarchivepkg.RegisterArchivedWorkflowServiceHandlerFromEndpoint, ctx, gwmux, endpoint, dialOpts)
 	mustRegisterGWHandler(clusterwftemplatepkg.RegisterClusterWorkflowTemplateServiceHandlerFromEndpoint, ctx, gwmux, endpoint, dialOpts)
 
-	mux.HandleFunc("/api/", func(w http.ResponseWriter, r *http.Request) { webhookInterceptor(w, r, gwmux) })
+	apiHandler := http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		webhookInterceptor(rw, r, gwmux)
+	})
+
+	if os.Getenv("ENABLE_SSE_KEEPALIVE") != "" {
```

Comment:
Initially, I figured it might be good to stick this behind a feature flag for people to try to make sure we don't blow anything up prior to rolling it out generally.